core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3442,7 +3442,15 @@ private MasterAuditContext createAuditContext(String command, AlluxioURI srcPath
       AlluxioURI dstPath, Inode srcInode) throws AccessControlException {
     MasterAuditContext auditContext = new MasterAuditContext(mAsyncAuditLogWriter);
     if (mAsyncAuditLogWriter != null) {
-      String ugi = AuthenticatedClientUser.getClientUser();
+      String user = AuthenticatedClientUser.getClientUser();
+      String ugi;
+      try {
+        String primaryGroup = CommonUtils.getPrimaryGroupName(user);
```

Comment:
`CommonUtils. getPrimaryGroupName` might not be a cheap operation. But since we have the `CachedGroupMapping`, and `PermissionChecker` also uses this for each permission check (a.k.a each metadata op), this should be okay for now. It would be good to see the performance overhead introduced by Audit logs.